etcd3/stateful/watch.py

```diff
@@ -238,7 +240,28 @@ def onEvent(self, filter_or_cb, cb=None):
             cb = filter_or_cb
         if not callable(cb):
             raise TypeError('callback should be a callable')
-        self.callbacks.append((self.get_filter(filter), filter, cb))
+        filter_func = self.get_filter(filter)
+        with self.callbacks_lock:
+            self.callbacks.append((filter_func, filter, cb))
+
+    @check_param(at_least_one_of=['filter', 'cb'])
+    def unEvent(self, filter=None, cb=None): # noqa # ignore redefinition of filter
+        """
+        remove a callback or filter event that's been previously added via onEvent()
+        If both parameters are given they are ANDd together; to OR the, make two calls.
+
+        :type filter: callable or regex string or EventType
+        :param filter: the callable filter or regex string or EventType the event to be removed was registerd with
+        :param cb: the callback funtion the event to be removed was registerd with
+        """
+        with self.callbacks_lock:
+            for i in reversed(range(len(self.callbacks))):
```

Comment:
still not getting the point of "iterating over inverted indices", but that's okay

Reply:
if you need to delete items 3 and 4, and you delete `itemlist[3]`, and then `itemlist[4]` then you've really deleted items 3 and 5, because deleting item 3 renumbers all the items after 3.  If you delete `itemlist[4]` and then `itemlist[3]` then you remove items 3 and 4.

Reply:
Get it